core/src/data/clientGeoJsonSource.cpp

```diff
@@ -0,0 +1,186 @@
+
+    auto feature = geojsonvt::Convert::create(geojsonvt::Tags(),
+                                              geojsonvt::ProjectedFeatureType::Point,
+                                              container.members);
+
+    m_features.push_back(feature);
+    m_store = std::make_unique<GeoJSONVT>(m_features, maxZoom, indexMaxZoom, indexMaxPoints, tolerance);
+
+}
+
+void ClientGeoJsonSource::addLine(double* _coords, int _lineLength) {
+    
+    std::vector<geojsonvt::LonLat> line(_lineLength, { 0, 0 });
+    for (int i = 0; i < _lineLength; i++) {
+        line[i] = { _coords[2 * i], _coords[2 * i + 1] };
+    }
+    
+    std::vector<geojsonvt::ProjectedGeometry> geometry = { geojsonvt::Convert::project(line, tolerance) };
+
+    auto feature = geojsonvt::Convert::create(geojsonvt::Tags(),
+                                              geojsonvt::ProjectedFeatureType::LineString,
+                                              geometry);
+    
+    m_features.push_back(std::move(feature));
+    m_store = std::make_unique<GeoJSONVT>(m_features, maxZoom, indexMaxZoom, indexMaxPoints, tolerance);
+}
+
+void ClientGeoJsonSource::addPoly(double* _coords, int* _ringLengths, int _rings) {
+
+    geojsonvt::ProjectedGeometryContainer geometry;
+    double* ringCoords = _coords;
+    for (int i = 0; i < _rings; i++) {
+        int ringLength = _ringLengths[i];
+        std::vector<geojsonvt::LonLat> ring(ringLength, { 0, 0 });
+        for (int j = 0; j < ringLength; j++) {
+            ring[j] = { ringCoords[2 * j], ringCoords[2 * j + 1] };
+        }
+        geometry.members.push_back(geojsonvt::Convert::project(ring, tolerance));
+        ringCoords += 2 * ringLength;
+    }
+
+    auto feature = geojsonvt::Convert::create(geojsonvt::Tags(),
+                                              geojsonvt::ProjectedFeatureType::Polygon,
+                                              geometry);
+
+    m_features.push_back(feature);
+    m_store = std::make_unique<GeoJSONVT>(m_features, maxZoom, indexMaxZoom, indexMaxPoints, tolerance);
+}
+
+std::shared_ptr<TileData> ClientGeoJsonSource::parse(const Tile& _tile, std::vector<char>& _rawData) const {
+
+    if (!m_store) { return nullptr; }
+
+    auto data = std::make_shared<TileData>();
+
+    auto id = _tile.getID();
+
+    auto tile = m_store->getTile(id.z, id.x, id.y); // uses a mutex lock internally for thread-safety
+
+    Layer layer(""); // empty name will skip filtering by 'collection'
+
+    for (auto& it : tile.features) {
+
+        Feature feat;
+
+        const auto& geom = it.tileGeometry;
+        const auto type = it.type;
+
+        switch (type) {
+            case geojsonvt::TileFeatureType::Point: {
+                feat.geometryType = GeometryType::points;
+                for (const auto& pt : geom) {
+                    const auto& point = pt.get<geojsonvt::TilePoint>();
+                    feat.points.push_back(transformPoint(point));
```

Comment:
ohh yes ... didn't see the implementation for that when I was checking this .. my bad.